### PR TITLE
[FIX] web_editor, website: hide image animation option when image-url

### PR DESCRIPTION
**Pull request by rdeodoo**
Before this commit, the "animation" option would be shown for image of
type url-attachment-redirect.
From there, selection the option would result in a crash if the image
was CORS protected (which is often the case).

Step to reproduce:
- Drag & drop "Text - Image" snippet
- Double click on an image to replace it
- Select in the media dialog "Add URL" and insert a CORS protected
  image URL
- The image is correctly added, and its url is something like
  `/web/image/123-redirect/xxx.jpg`
- Click on the image and then click on its "Animation" option
-> It crashes

Same flow is possible with absolute URL of a CORS protected image.

Technical note: JS `fetch()` takes advantage of the browser cache, no
need to create a `Map` cache for it, despite the `_computeVisibility()`
method being called multiple times.

### CODE REVIEW

**Comment by rdeodoo** [comment]
As of today, one can reproduce with the following images:
- Cors protected: [here](https://tinyjpg.com/images/social/website.jpg)
- Not cors protected: [here](https://upload.wikimedia.org/wikipedia/commons/f/f9/Phoenicopterus_ruber_in_S%C3%A3o_Paulo_Zoo.jpg)

**Comment by rdeodoo** [comment]
@qsm-odoo This one is following https://github.com/odoo/odoo/commit/f0fe283c761cd6b1d7293dd41795ac6fc721e341 report

It's stuff I fixed while investigating the real fix here: https://github.com/odoo/odoo/pull/146731

I proposed two solutions for the cors protect images and the animation option:
1. Hide the animation option entirely 
2. Hide only the problematic animation option value (on hover)

Second is probably better, but since I started with 1.. I kept the two.

Maybe we could also go with something more user friendly like keeping the option but on click do nothing and just warn them that the image does not support the option and that they should try to download it and upload it instead of using link? Seems maybe overkill?

File: addons/web_editor/static/src/js/editor/image_processing.js [hunk 1]
```diff
@@ -511,6 +511,31 @@ export function isImageSupportedForStyle(img) {
...
+ * @param {HTMLImageElement} img
+ * @returns {Promise<Boolean>}
+ */
+export async function isImageCorsProtected(img) {
+    const src = img.getAttribute('src');
+    if (!src) {
+        return false;
+    }
+    let isCordProtected = false;
+    if (src && (!src.startsWith("/") || /\/web\/image\/\d+-redirect\//.test(src))) {
```
Comment (qsm-odoo) [inline addons/web_editor/static/src/js/editor/image_processing.js:524]: No need to check "src", you already did it above?

File: addons/web_editor/static/src/js/editor/image_processing.js [hunk 2]
```diff
@@ -511,6 +511,31 @@ export function isImageSupportedForStyle(img) {
...
+        // The `fetch()` used later in the code might fail if the image is
+        // CORS protected. We check upfront if it's the case.
+        // Two possible cases:
+        // 1. the `src` is an absolute URL from another domain.
+        //    For instance, abc.odoo.com vs abc.com which are actually the
+        //    same database behind.
+        // 2. A "attachment-url" which is just a redirect to the real image
+        //    which could be hosted on another website.
+        isCordProtected = await fetch(src)
```
Comment (qsm-odoo) [inline addons/web_editor/static/src/js/editor/image_processing.js:533]: Maybe you could try not fetching the whole image to have the info? Maybe via `fetch(url, {method: 'HEAD'})`?

**Review by qsm-odoo — CHANGES_REQUESTED** [review]
@rdeodoo I would only hide the option "hover", the 2 others are basically only animating the img node, not its source so... :+1:

File: addons/web_editor/static/src/js/editor/image_processing.js (hunk 1 above)
Comment (rdeodoo) [reply addons/web_editor/static/src/js/editor/image_processing.js:524]: Whoops indeed

File: addons/web_editor/static/src/js/editor/image_processing.js (hunk 2 above)
Comment (rdeodoo) [reply addons/web_editor/static/src/js/editor/image_processing.js:533]: Very good catch :+1:

**Comment by rdeodoo** [comment]
Perfect then, it should be ready :+1:

**Review by qsm-odoo — APPROVED** [review]
@robodoo rebase-ff r+

**Pull request closed without merging**